code/game/machinery/constructable_frame.dm

```diff
@@ -658,6 +658,33 @@ to destroy them and players will be able to make replacements.
 							/obj/item/stock_parts/manipulator = 2,
 							/obj/item/reagent_containers/glass/beaker = 2)
 
+/obj/item/circuitboard/dish_drive
+	board_name = "Dish Drive"
+	build_path = /obj/machinery/dish_drive
+	board_type = "machine"
+	origin_tech = "programming=2"
+	var/suction = TRUE
+	var/transmit = TRUE
+	req_components = list(
+							/obj/item/stock_parts/manipulator = 1,
+							/obj/item/stock_parts/matter_bin = 1,
+							/obj/item/stack/sheet/glass = 1)
+
+/obj/item/circuitboard/dish_drive/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Its suction function is [suction ? "enabled" : "disabled"]. Use it in-hand to switch.</span>"
+	. += "<span class='notice'>Its disposal auto-transmit function is [transmit ? "enabled" : "disabled"]. Alt-click it to switch.</span>"
+
+/obj/item/circuitboard/dish_drive/attack_self(mob/living/user)
+	suction = !suction
+	to_chat(user, "<span class='notice'>You [suction ? "enable" : "disable"] the board's suction function.</span>")
+
+/obj/item/circuitboard/dish_drive/AltClick(mob/living/user)
```

Comment:
can you check to make sure this doesn't allow observers to interact with this?

Reply:
Checking...

Reply:
Yeah, ghost and observers cant interact with it.